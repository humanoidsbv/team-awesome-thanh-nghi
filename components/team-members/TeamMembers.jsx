import PropTypes from 'prop-types';
import React from 'react';
import Link from 'next/link';

import './team-members.scss';

import Main from '../../shared/components/Main';
import PageHeader from '../../shared/components/PageHeader';
import Select from '../../shared/components/Select';
import TeamMember from '../team-member';

import { dateToLocaleStringMonthYear } from '../../shared/services/converter-time';

class TeamMembers extends React.Component {
  handleChange = ({ target }) => {
    const { sortTeamMembers } = this.props;
    sortTeamMembers(target.value);
  }

  render() {
    const { teamMembers } = this.props;
    return (
      <React.Fragment>
        <PageHeader
          title="Team Members"
          subtitle="5 Members"
        />
        <Main>
          <div className="members-header">
            <h2 className="members-header__title">All Humanoids</h2>
            <Link href="/new-team-member">
              <button
                className="members-header__add-button"
                type="button"
              >
                <img
                  alt="Plus icon"
                  className="members-header__add-icon"
                  src="/static/images/icon-plus.svg"
                />
                New Humanoid
              </button>
            </Link>
            <Select
              className="select-element__sort"
              defaultValue={{ label: 'Sort by:', value: '' }}
              onChange={this.handleChange}
              options={[
                {
                  label: 'First name A-Z',
                  value: 'firstName-ascending'
                },
                {
                  label: 'First name Z-A',
                  value: 'firstName-descending'
                },
                {
                  label: 'Last name A-Z',
                  value: 'lastName-ascending'
                },
                {
                  label: 'Last name Z-A',
                  value: 'lastName-descending'
                },
                {
                  label: 'Employee Number 0-9',
                  value: 'memberNumber-ascending'
                },
                {
                  label: 'Employee Number 9-0',
                  value: 'memberNumer-descending'
                },
                {
                  label: 'Current Employer A-Z',
                  value: 'clientName-ascending'
                },
                {
                  label: 'Current Employer Z-A',
                  value: 'clientName-descending'
                },
                {
                  label: 'Starting Date Old-New',
                  value: 'startDate-ascending'
                },
                {
                  label: 'Starting Date New-Old',
                  value: 'startDate-descending'
                }
              ]}
              required
            />
          </div>
          <div className="members-list">
            {teamMembers.map((teamMember) => {
              const localizedStartDate = dateToLocaleStringMonthYear(teamMember.startDate);
              return (
                <React.Fragment key={teamMember.id}>
                  <TeamMember
                    currentClient={teamMember.clientName}
                    description={teamMember.description}
                    firstName={teamMember.firstName}
                    lastName={teamMember.lastName}
                    memberNumber={teamMember.memberNumber}
                    startDate={localizedStartDate}
                  />
                </React.Fragment>
              );
            })}
          </div>
        </Main>
      </React.Fragment>
    );
  }
}


TeamMembers.propTypes = {
  sortTeamMembers: PropTypes.func.isRequired,
  teamMembers: PropTypes.arrayOf(PropTypes.shape({})).isRequired
};

export default TeamMembers;
