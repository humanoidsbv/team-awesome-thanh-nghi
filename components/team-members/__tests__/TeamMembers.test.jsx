import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import TeamMembers from '../TeamMembers';

const retrieveClients = () => {};
const retrieveTeamMembers = () => {};
const sortTeamMembers = () => {};

const teamMembers = [
  {
    clientName: 'Humanoids',
    currentClient: '3',
    firstName: 'Pietje',
    lastName: 'Puk',
    description: 'Front End Developer',
    memberNumber: 'H-007',
    address: 'Goudseweg 12',
    zipcode: '6857SK',
    city: 'Verweggiestad',
    emailAddress: 'pietje.puk@humanoids.nl',
    linkedIn: 'linkedin.com/pietje-puk',
    startDate: '2019-03-03T15:07:44.742Z',
    id: 'aSMiO9N'
  },
  {
    clientName: 'Port of Rotterdam',
    currentClient: '2',
    firstName: 'Jantje',
    lastName: 'Smit',
    description: 'Designer',
    memberNumber: 'H-008',
    address: 'Zilverstraat 8',
    zipcode: '1234GS',
    city: 'Timboektoe',
    emailAddress: 'jantje.smit@humanoids.nl',
    linkedIn: 'linkedin.com/jantje-smit',
    startDate: '2019-04-03T15:07:44.742Z',
    id: 'KfAM5Sn'
  }
];

const clientNameIdSelector = [
  {
    client: 'Port of Rotterdam',
    id: '1'
  },
  {
    client: 'Port of Rotterdam',
    id: '2'
  },
  {
    client: 'Humanoids',
    id: '3'
  }
];

it('renders a snapshot', () => {
  const wrapper = shallow(
    <TeamMembers
      clients={clientNameIdSelector}
      onRetrieveClients={retrieveClients}
      onRetrieveTeamMembers={retrieveTeamMembers}
      sortTeamMembers={sortTeamMembers}
      teamMembers={teamMembers}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
