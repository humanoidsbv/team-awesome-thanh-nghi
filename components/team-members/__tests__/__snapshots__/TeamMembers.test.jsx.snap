// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<Fragment>
  <PageHeader
    subtitle="2 Members"
    title="Team Members"
  />
  <Container>
    <div
      className="members-header"
    >
      <h2
        className="members-header__title"
      >
        All Humanoids
      </h2>
      <Link
        href="/new-team-member"
      >
        <button
          className="members-header__add-button"
          type="button"
        >
          <img
            alt="Plus icon"
            className="members-header__add-icon"
            src="/static/images/icon-plus.svg"
          />
          New Humanoid
        </button>
      </Link>
      <Select
        className="select-element__sort"
        defaultValue={
          Object {
            "label": "Sort by:",
            "value": "",
          }
        }
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "First name A-Z",
              "value": "firstName-ascending",
            },
            Object {
              "label": "First name Z-A",
              "value": "firstName-descending",
            },
            Object {
              "label": "Last name A-Z",
              "value": "lastName-ascending",
            },
            Object {
              "label": "Last name Z-A",
              "value": "lastName-descending",
            },
            Object {
              "label": "Employee Number 0-9",
              "value": "memberNumber-ascending",
            },
            Object {
              "label": "Employee Number 9-0",
              "value": "memberNumer-descending",
            },
            Object {
              "label": "Current Employer A-Z",
              "value": "clientName-ascending",
            },
            Object {
              "label": "Current Employer Z-A",
              "value": "clientName-descending",
            },
            Object {
              "label": "Starting Date Old-New",
              "value": "startDate-ascending",
            },
            Object {
              "label": "Starting Date New-Old",
              "value": "startDate-descending",
            },
          ]
        }
        required={true}
      />
    </div>
    <div
      className="members-list"
    >
      <TeamMember
        currentClient="Humanoids"
        description="Front End Developer"
        firstName="Pietje"
        key="aSMiO9N"
        lastName="Puk"
        memberNumber="H-007"
        startDate="March 2019"
      />
      <TeamMember
        currentClient="Port of Rotterdam"
        description="Designer"
        firstName="Jantje"
        key="KfAM5Sn"
        lastName="Smit"
        memberNumber="H-008"
        startDate="April 2019"
      />
    </div>
  </Container>
</Fragment>
`;
